### ISSUE TITLE
L.DomUtil.disableTextSelection should only disable text selection on the map container

### ISSUE THREAD

**Issue by ventralnet** [Leaflet/Leaflet#9259]
### Checklist

- [X] I've looked at the [documentation](https://leafletjs.com/reference.html) to make sure the behavior isn't documented and expected.
- [X] I'm sure this is an issue with Leaflet, not with my app or other dependencies (Angular, Cordova, React, etc.).
- [X] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this hasn't been reported yet.
- [X] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Steps to reproduce

I have a certain situation in an application where we have a content editable div as a search bar. The search bar is special in that we highlight special search terms as the user types. To keep track of cursor positions as the user types we deal with text selection of the document.

When you enable a draw tool and the hooks are added there is a line that L.DomUtil.disableTextSelection. This prevents the special content editable div from receiving text select start events.

I think this is a bug. I think text selection should only be disabled on the map itself and not the entire DOM document.

This also may be a bug with L.DomUtil.disableTextSelection for leaflet

### Expected behavior

Text select is disabled on the map

### Current behavior

text selection is disabled on the entire window

### Environment

- Leaflet version: 
- Browser (with version): 
- OS/Platform (with version):

**Comment by IvanSanchez** [comment]
This is certainly doable.

The `disableTextSelection`/`enableTextSelection` functions in `DomUtil` should receive a HTMLElement as a parameter, and disable/enable text selection in that element only. If the parameter is `undefined`, then fall back to the document (this should prevent plugins from misbehaving).

Then, send the map's container as a parameter on calls to `disable`/`enableTextSelection`

**Comment by ventralnet** [comment]
@IvanSanchez I agree.  I was surprised looking through the docs it did not.  I could probably submit a pull request sometime this week when I have time

**Comment by mourner** [comment]
I think it was designed this way because you're not meant to disable it permanently, only for the duration of a particular drag action. This is what Leaflet does, disabling it on the whole document because dragging cursor can go outside of the map container while continuing to affect the map, and we don't want text to be selected at the time. This seems like a bug in the draw tool you're using rather than a Leaflet issue — correct me if I'm wrong.

**Comment by ventralnet** [comment]
@mourner This makes sense.  I was conflicted if it were a leaflet bug or a Leaflet.draw bug actually.  My work around is I cancel all leaflet draw tools when the leaflet map is blurred which in turn calls L.DomUtil.enableTextSelection

**Comment by PrakharSingh42** [comment]
@ventralnet can i try this i wanna make my first contribution

**Comment by ventralnet** [comment]
> [@ventralnet](https://github.com/ventralnet) can i try this i wanna make my first contribution

Hey! Contributing is great for any open source project.  I  am not a maintainer of leaflet though, so you would want to ask one of them. I just reported the issue

**Comment by IvanSanchez** [comment]
@ventralnet @PrakharSingh42 No need to ask for permission, just read the whole conversation, fix it, **test it in different conditions and web browsers**, and send a PR.

**Comment by Duly330AI** [comment]
I implemented a scoped fix and added tests. PR: https://github.com/Leaflet/Leaflet/pull/9960

Quick repro snippet you can try in the console on any page with a Leaflet map:

```js
// disable selection only on the map container
L.DomUtil.disableTextSelection(map.getContainer());
// selection still works on other elements like a contenteditable div
document.querySelector('#myEditable').focus();
```

What I changed: make `disableTextSelection(el)`/`enableTextSelection(el)` accept an optional container element, updated Draggable/BoxZoom call sites, restored `disableImageDrag()`, and added `spec/suites/dom/DomUtilSpec.js` to test scoped behavior.

Tests: I ran the full suite locally and all tests passed (1067).

**Comment by Duly330AI** [comment]
I implemented a scoped fix and added tests. PR: https://github.com/Leaflet/Leaflet/pull/9960

Quick repro snippet you can try in the console on any page with a Leaflet map:

```js
// disable selection only on the map container
L.DomUtil.disableTextSelection(map.getContainer());
// selection still works on other elements like a contenteditable div
document.querySelector('#myEditable').focus();
```

What I changed: make `disableTextSelection(el)`/`enableTextSelection(el)` accept an optional container element, updated Draggable/BoxZoom call sites, restored `disableImageDrag()`, and added `spec/suites/dom/DomUtilSpec.js` to test scoped behavior.

Tests: I ran the full suite locally and all tests passed (1067).

**Comment by mourner** [comment]
> I think it was designed this way because you're not meant to disable it permanently, only for the duration of a particular drag action. This is what Leaflet does, disabling it on the whole document because dragging cursor can go outside of the map container while continuing to affect the map, and we don't want text to be selected at the time. This seems like a bug in the draw tool you're using rather than a Leaflet issue — correct me if I'm wrong.

Quiting this again as the rationale to not do this and closing.
